golem/network/hyperdrive/client.py

```diff
@@ -76,7 +77,7 @@ def add(self, files, client_options=None, **kwargs):
             command='upload',
             id=kwargs.get('id'),
             files=files,
-            timeout=round_timeout(client_options.timeout)
+            timeout=round_timeout(client_options.timeout + GRACE_PERIOD)
```

Comment:
```suggestion
            timeout=round_timeout(client_options.timeout) + GRACE_PERIOD
```
A more meaningful error message will be raised when `timeout` is `None`.